src/cpu.rs

```diff
@@ -117,15 +151,36 @@ mod tests {
 
     #[test]
     fn do_instruction() {
-        let add_immediate = Instruction::AddSub {
+        let add_register = Instruction::AddSub {
             opcode: OpcodeAddSub::AddRegister,
             operand: 7,
-            source_register: 1,
+            source_register: 10,
             destination_register: 0,
         };
+        let sub_register = Instruction::AddSub {
+            opcode: OpcodeAddSub::SubRegister,
+            operand: 7,
+            source_register: 10,
+            destination_register: 1,
+        };
+        let add_immediate = Instruction::AddSub {
+            opcode: OpcodeAddSub::AddImmediate,
+            operand: 7,
+            source_register: 10,
+            destination_register: 2,
+        };
+        let sub_immediate = Instruction::AddSub {
+            opcode: OpcodeAddSub::SubImmediate,
+            operand: 1,
+            source_register: 10,
+            destination_register: 3,
+        };
 
         let mut cpu = ARM7TDMI::new();
+        cpu.execute_instruction(add_register);
+        cpu.execute_instruction(sub_register);
         cpu.execute_instruction(add_immediate);
+        cpu.execute_instruction(sub_immediate);
 
         println!("{:?}", cpu);
```

Comment:
Could you add asserts for the registers as opposed to printing the values?
That way we can ensure the operations did the correct thing, as well as being able to accept the instructions in general.